src/ae.h

```diff
@@ -1,139 +1,19 @@
-         READABLE event already fired in the same event                                                                \
-         loop iteration. Useful when you want to persist                                                               \
-         things to disk before sending replies, and want                                                               \
-         to do that in a group fashion. */
-
-#define AE_FILE_EVENTS (1 << 0)
-#define AE_TIME_EVENTS (1 << 1)
-#define AE_ALL_EVENTS (AE_FILE_EVENTS | AE_TIME_EVENTS)
-#define AE_DONT_WAIT (1 << 2)
-#define AE_CALL_BEFORE_SLEEP (1 << 3)
-#define AE_CALL_AFTER_SLEEP (1 << 4)
-
-#define AE_NOMORE -1
-#define AE_DELETED_EVENT_ID -1
-
-/* Macros */
-#define AE_NOTUSED(V) ((void)V)
-
-struct aeEventLoop;
-
-/* Types and data structures */
-typedef void aeFileProc(struct aeEventLoop *eventLoop, int fd, void *clientData, int mask);
-typedef int aeTimeProc(struct aeEventLoop *eventLoop, long long id, void *clientData);
-typedef void aeEventFinalizerProc(struct aeEventLoop *eventLoop, void *clientData);
-typedef void aeBeforeSleepProc(struct aeEventLoop *eventLoop);
-typedef void aeAfterSleepProc(struct aeEventLoop *eventLoop, int numevents);
-
-/* File event structure */
-typedef struct aeFileEvent {
-    int mask; /* one of AE_(READABLE|WRITABLE|BARRIER) */
-    aeFileProc *rfileProc;
-    aeFileProc *wfileProc;
-    void *clientData;
-} aeFileEvent;
-
-/* Time event structure */
-typedef struct aeTimeEvent {
-    long long id; /* time event identifier. */
-    monotime when;
-    aeTimeProc *timeProc;
-    aeEventFinalizerProc *finalizerProc;
-    void *clientData;
-    struct aeTimeEvent *prev;
-    struct aeTimeEvent *next;
-    int refcount; /* refcount to prevent timer events from being
-                   * freed in recursive time event calls. */
-} aeTimeEvent;
+#include "ae_eventloop.h"
 
-/* A fired event */
-typedef struct aeFiredEvent {
-    int fd;
-    int mask;
-} aeFiredEvent;
 
-/* State of an event based program */
-typedef struct aeEventLoop {
-    int maxfd;   /* highest file descriptor currently registered */
-    int setsize; /* max number of file descriptors tracked */
-    long long timeEventNextId;
-    aeFileEvent *events; /* Registered events */
-    aeFiredEvent *fired; /* Fired events */
-    aeTimeEvent *timeEventHead;
-    int stop;
-    void *apidata; /* This is used for polling API specific data */
-    aeBeforeSleepProc *beforesleep;
-    aeAfterSleepProc *aftersleep;
-    int flags;
-} aeEventLoop;
+/* This header file declares functions that are implemented in ae.c.
+ * These functions may alter the state of the underlying apidata,
+ * which may require a mutex for modification in the case of the server
+ * main event loop (`server.el`), as it may be accessed by IO threads.
+ * In files that may access the main event loop, we will include only the
+ * eventloop.h header file, which does not include the functions declared here. */
```

Comment:
AE does not depend on server and it is used in valkey-benchmark too, and it can also be used in other projects. We can avoid describing AE in terms of server.

```suggestion
 * These functions may alter the state of the underlying apidata,
 * which may require a mutex for modification if accessed by multiple threads.
 * Files that don't need to access the event loop apidata can instead include the
 * ae_eventloop.h header file, which does not include the functions declared here. */
```